pkg/cmd/repo/edit/edit.go

```diff
@@ -403,16 +408,26 @@ func interactiveRepoEdit(opts *EditOptions, r *api.Repository) error {
 				return err
 			}
 		case optionVisibility:
-			opts.Edits.Visibility = &r.Visibility
-			//nolint:staticcheck // SA1019: prompt.SurveyAskOne is deprecated: use Prompter
-			err = prompt.SurveyAskOne(&survey.Select{
-				Message: "Visibility",
-				Options: []string{"public", "private", "internal"},
-				Default: strings.ToLower(r.Visibility),
-			}, opts.Edits.Visibility)
+			opts.Edits.Visibility = new(string)
+			*opts.Edits.Visibility = r.Visibility
+
+			visibilityOptions := []string{"public", "private", "internal"}
+			selected, err := opts.Prompter.Select("Visibility", strings.ToLower(r.Visibility), visibilityOptions)
 			if err != nil {
 				return err
 			}
+			*opts.Edits.Visibility = visibilityOptions[selected]
+
+			if *opts.Edits.Visibility == "private" {
+				cs := opts.IO.ColorScheme()
+				fmt.Fprintf(opts.IO.ErrOut, "%s Changing the repository visibility to private will cause permanent loss of stars and watchers.\n", cs.WarningIcon())
```

Comment:
Wild thought: printing this warning and showing the confirmation prompt adds no value if the repository has no watchers and no stars (apart from the repo owner themselves). Would we only show this for repositories that have some meaningful number of watchers and stars?

Reply:
I added this functionality and made the prompt level anything more than 0 stars or 0 watchers.